sdk/python/tests/compiler/testdata/artifact_location.yaml

```diff
@@ -14,6 +14,11 @@
 apiVersion: argoproj.io/v1alpha1
 kind: Workflow
 metadata:
+  annotations:
+    kubeflow.org/pipelines/pipeline_spec: '{"name": "foo", "description": "hello world",
+      "inputs": [{"name": "tag", "description": "", "type": "", "default": null},
```

Comment:
In case of components, it was much simpler, with key value pairs...

metadata:
  annotations: {platform: 'OpenSource'}

Any reason to use this kubeflow.org/pipelines/pipeline_spec?

Reply:
> Any reason to use this kubeflow.org/pipelines/pipeline_spec?

We want to "namespace" the parameter so that it never clashes with user-provided annotations. Argo does the same.